src/main/cpp/startup_options.h

```diff
@@ -142,6 +142,11 @@ class StartupOptions {
   // Blaze's output base.  Everything is relative to this.  See
   // the BlazeDirectories Java class for details.
   blaze_util::Path output_base;
+  std::string output_base_option;
+
+  // Suffix to append to Blaze's output base, e.g. `-extra` or `/../no-hash`.
```

Comment:
Could you change "append" to "path-append" to make it clear the resulting string will be post-processed into a valid normalized and absolute path?

Reply:
Oh, and we should absolutely test for these `/../no-hash` up-references in the tests for this 😄